frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -73,134 +75,197 @@ const getOpenInWindowCallback = (url: string) => (): void => {
   window.open(url, "_blank")
 }
 
-function MainMenu(props: Props): ReactElement {
-  const [dropdownOpen, setDropdownOpen] = useState(false)
-  const isServerDisconnected = !props.isServerConnected()
-
-  function toggleDropdown(): void {
-    setDropdownOpen(prevState => !prevState)
-  }
+export interface MenuListItemProps {
+  item: any
+  "aria-selected": boolean
+  onClick: (e: MouseEvent<HTMLLIElement>) => void
+  onMouseEnter: (e: MouseEvent<HTMLLIElement>) => void
+  $disabled: boolean
+  $isHighlighted: boolean
+}
 
-  const S4AMenuOptions = props.s4aMenuItems.map(item => {
-    if (item.type === "separator") {
-      return <DropdownItem divider />
-    }
+const MenuListItem = forwardRef<HTMLLIElement, MenuListItemProps>(
```

Comment:
The purpose of MenuListItem isn't clear (at least to me) - can you add a comment above this explaining what it does?

Reply:
Yes. I'll add something.